cfpb/sbl-frontend
# [component] Error500 - Server error

Closes #160 

## How it works
Huge thanks to @shindigira for the recommendation to use react-router-dom's `useNavigate` to pass error state. 

- When we encounter a Server error:
```
import { useNavigate } from 'react-router-dom';

const navigate = useNavigate()
navigate('/500', {state: {errorMessage: 'The sky is falling...', statusCode: '500'}}); 
```

- On the `/500` page, for MVP-only, we display a `Show Error details` button that, upon click, will show the error `statusCode` and `errorMessage` passed via `navigate(...)`

## Prior to merge
- [x] Design team review
- [x] Dev team review
- [x] Remove [example](https://github.com/cfpb/sbl-frontend/pull/161/files#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2R33-R43) error

## Changes

- Creates `<Error500>` component
- Copies styles and assets from CF.gov
- Updates styles per Design feedback to match `light` theme and adjust spacing


## How to test this PR

1. Visit http://localhost:8899/500
2. Observe no `Show error details` button (No error)
1. Visit http://localhost:8899/500/demo
4. Click `Show error details` (Error hidden)
5. See that the sky is falling (Error visible)

## Screenshots
### Desktop
![500-desktop](https://github.com/cfpb/sbl-frontend/assets/2592907/64e8d6a3-9605-4266-9e3d-aede85505340)

### Mobile
![500-mobile](https://github.com/cfpb/sbl-frontend/assets/2592907/be3cd132-1fb9-43c4-9e0a-9af859503a38)
